packages/flutter/lib/src/services/autofill.dart

```diff
@@ -0,0 +1,153 @@
+import 'package:flutter/foundation.dart';
+
+import 'text_input.dart';
+
+@immutable
+class AutofillConfiguration {
+  const AutofillConfiguration({
+    @required this.uniqueIdentifier,
+    @required this.autofillHints,
+    this.currentEditingValue,
+  }) : assert(uniqueIdentifier != null),
+       assert(autofillHints != null);
+
+  final String uniqueIdentifier;
+  final List<String> autofillHints;
+  final TextEditingValue currentEditingValue;
+
+  /// Returns a representation of this object as a JSON object.
+  Map<String, dynamic> toJson() {
+    return <String, dynamic>{
+      'uniqueIdentifier': uniqueIdentifier,
+      'hints': autofillHints,
+      'editingValue': currentEditingValue.toJSON(),
+    };
+  }
+}
+
+// Ideally AutofillClient should NOT extend TextInputClient. Rather, TextInputClient
+// should have an `AutofillDelegate` property.
+abstract class AutofillClient extends TextInputClient {
+  TextInputConfiguration get textInputConfiguration;
```

Comment:
as discussed offline, may be for the first milestone we can have each textfield's build send their own information to platform, without waiting on an open connection. That way we can omit the use of TextInputClient.